query/src/org/labkey/query/QueryServiceImpl.java

```diff
@@ -264,6 +271,125 @@ public WhereClause createFilterClause(@NotNull FieldKey fieldKey, Object value)
         }
     };
 
+    private static SQLFragment getColumnInSql(@NotNull FieldKey fieldKey, Set<String> params, User user, Container container, Map<FieldKey, ? extends ColumnInfo> columnMap, boolean negate)
+    {
+        ColumnInfo col = columnMap.get(fieldKey);
+        SQLFragment colFrag = new SQLFragment(col == null ? fieldKey.toString() : col.getAlias());
+        String[] values = params.toArray(new String[0]);
+
+        if (values.length < 3)
+            throw new NotFoundException("Incorrect number of arguments provided");
+
+        String selectColumn = values[0];
+        String schemaName = values[1];
+        String queryName = values[2];
+        String whereSql = null;
+
+        if (values.length > 3)
+            whereSql = values[3];
+
+        if (StringUtils.isEmpty(selectColumn))
+            throw new NotFoundException("Select column not specified");
+        if (StringUtils.isEmpty(schemaName))
+            throw new NotFoundException("SchemaName not specified");
+        if (StringUtils.isEmpty(queryName))
+            throw new NotFoundException("QueryName not specified");
+
+        if (user == null || container == null)
+            throw new NotFoundException("Invalid context");
+
+        QuerySchema querySchema = DefaultSchema.get(user, container, schemaName);
+        if (!(querySchema instanceof UserSchema userSchema))
+            throw new NotFoundException("Could not find the specified schema in the folder '" + container.getPath() + "'");
+
+        SqlDialect dialect = querySchema.getDbSchema().getSqlDialect();
+        SQLFragment selectIdsSql = new SQLFragment()
+                .append("SELECT ")
+                .append(dialect.makeLegalIdentifier(selectColumn))
+                .append(" FROM ")
+                .append(dialect.makeLegalIdentifier(schemaName))
+                .append(".")
+                .append(dialect.makeLegalIdentifier(queryName));
+        if (!StringUtils.isEmpty(whereSql))
+        {
+            selectIdsSql = selectIdsSql.append(" WHERE (")
+                    .append(whereSql)
+                    .append(")");
+        }
+
+        QueryService qs = QueryService.get();
+        QueryDefinition qd;
+        qd = qs.createQueryDef(user, container, userSchema, GUID.makeGUID().replace("-", ""));
```

Comment:
Need the separate initialization?
```suggestion
        QueryDefinition qd = qs.createQueryDef(user, container, userSchema, GUID.makeGUID().replace("-", ""));
```